src/tool/xmeta/cpp/xmeta_idl_reader.cpp

```diff
@@ -43,52 +35,72 @@ namespace xlang::xmeta
         return parser.getNumberOfSyntaxErrors();
     }
 
+    struct invalid_ns_id
+    {
+        invalid_ns_id(std::string_view const& name) : new_name{ name } { }
+        bool operator()(std::pair<std::string_view, std::shared_ptr<namespace_model>> const& v) const
+        {
+            auto old_name = v.first;
+#if defined(_WIN32)
```

Comment:
If this method only works correctly on Windows, and we don't have an alternate implementation for other platforms, then only build this binary for Windows (instead of building misbehaving components for other platforms). Update CMakelists.txt accordingly.